File: java/java-arithmetic/src/main/java/com/leetcode/first/J35.java
package com.leetcode.first;

public class J35 {

    public static void main(String[] args) {
        int i = new J35().searchInsert(new int[]{1, 3, 5, 6}, 7);
        System.out.println(i);
    }

    public int searchInsert(int[] nums, int target) {
        int left=0;
        int right=nums.length-1;
        int ans=nums.length;
        while (left<=right){
            int mid=left+(right-left)/2;
            if (nums[mid]<target){
                left=mid+1;
            }
            if (nums[mid]==target){
                return mid;
            }
            if (nums[mid]>target){
                ans=mid;
                right=mid-1;
            }
        }
        return ans;
    }
}

File: python-test/leetcode/J35.py
class Solution:
    def searchInsert(self, nums: list[int], target: int) -> int:
        n = len(nums)
        left=0
        right = n-1
        ans = n
        while left<=right:
            mid = left +(right-left)//2
            if nums[mid] < target:
                left=mid+1
            else:
                ans = mid
                right = mid-1
        
        return ans


print(Solution().searchInsert([1,3,5,6],7))